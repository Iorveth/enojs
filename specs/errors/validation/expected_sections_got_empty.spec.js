const eno = require('../../../eno.js');

const input = `
languages:
`.trim();

describe('validation.expectedSectionsGotEmpty', () => {
  const document = eno.parse(input);

  let error;
  try {
    document.sections('languages');
  } catch(err) {
    error = err;
  }

  it(`provides a correct message`, () => {
    expect(error.message).toMatchSnapshot();
  });

  it(`provides correct selection metadata`, () => {
    expect(error.selection).toMatchSnapshot();
  });
});
