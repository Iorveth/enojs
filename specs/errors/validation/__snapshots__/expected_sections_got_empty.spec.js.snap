// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation.expectedSectionsGotEmpty provides a correct message 1`] = `
"Only sections with the name 'languages' were expected, but an empty element with this name was found.

   Line | Content
 >    1 | languages:
"
`;

exports[`validation.expectedSectionsGotEmpty provides correct selection metadata 1`] = `
Array [
  Array [
    0,
    0,
  ],
  Array [
    0,
    10,
  ],
]
`;
